diff --git a/dist/server/index.js b/dist/server/index.js
index 1e56a6c..b323b05 100644
--- a/dist/server/index.js
+++ b/dist/server/index.js
@@ -59,15 +59,15 @@ import matter from "gray-matter";
 import { compile as compileMdx } from "@mdx-js/mdx";
 import remarkFrontmatter from "remark-frontmatter";
 var compile = async (content, options) => {
-  const compiled = await compileMdx(content, {
+  const compiled = await compileMdx({ value: content, path: options?.path }, {
     ...options ?? {},
     outputFormat: "function-body",
-    remarkPlugins: [remarkFrontmatter, ...options?.remarkPlugins ?? []]
-  });
+    remarkPlugins: [remarkFrontmatter, ...options?.remarkPlugins ?? []],
+  },);
   return String(compiled);
 };
 var getAttributes = (content) => {
-  const { data: attributes } = matter(content);
+  const { data: attributes, } = matter(content);
   return attributes;
 };
 
@@ -119,12 +119,13 @@ var loadMdx = async (request, options) => {
   const path = getFilePathBasedOnUrl(request.url, paths, aliases);
   const content = await getFileContent(path);
   const [mdxContent, attributes] = await Promise.all([
-    compile(content, options),
+    compile(content, { ...options, path }),
     getAttributes(content)
   ]);
   return {
     __raw: mdxContent,
-    attributes
+    attributes,
+    path
   };
 };
 var loadAllMdx = async (filterByPaths) => {
